### PR TITLE
Make meta data converter mapping generic and populate provisioner meta info

### PR DESCRIPTION
**Pull request by shobha2626**
1. **What this PR does / why we need it**: 

Make meta data converter mapping generic and populate provisioner meta info

Meta converter was not supporting the array format as that of a provisioner resource. As a result, meta was not getting populated for provisioner because the model path was not including the entire path. In Provisioner resource scenario the meta should be `provisioner.meta`.

Made changes to accomodate this change not just for provisioner but any resource that follows this structure

3. **Which issue(s) this PR fixes**

`
{
  "version": 4,
  "terraform_version": "1.6.4",
  "serial": 1,
  "lineage": "2dcd02f2-3dcc-d8a3-42b1-f05b03f9e9fe",
  "outputs": {},
  "resources": [
    {
      "mode": "data",
      "type": "tanzu-mission-control_provisioner",
      "name": "read_provisioner",
      "provider": "provider[\"vmware/devd/tanzu-mission-control\"]",
      "instances": [
        {
          "schema_version": 0,
          "attributes": {
            "id": "prvn:01HQ2DV4SAY0C5G832WHZ4TBST",
            "provisioners": [
              {
                "management_cluster": "eks",
                "meta": [
                  {
                    "annotations": {},
                    "description": "Create provisioner through terraform",
                    "labels": {
                      "key1": "value1",
                      "key2": "value2"
                    },
                    "resource_version": "4965",
                    "uid": "prvn:01HQ2DV4SAY0C5G832WHZ4TBST"
                  }
                ],
                "name": "mshobha-test",
                "org_id": ""
              }
            ]
          },
          "sensitive_attributes": []
        }
      ]
    }
  ],
  "check_results": null
}`

        (optional, in `fixes #<issue number>` format, will close the issue(s) when PR gets merged):

        Fixes #

4. **Additional information**

5. **Special notes for your reviewer**

### CODE REVIEW

**Review by ankitsny — APPROVED** [review]
LGTM

**Review by ankitsny — APPROVED** [review]

File: internal/resources/provisioner/provisioner_data_source_test.go [hunk 1]
```diff
@@ -84,12 +83,11 @@ func verifyProvisionerDataSource(name string) resource.TestCheckFunc {
...
-//	return []resource.TestCheckFunc{
-//		resource.TestCheckResourceAttrPair(dataSourceName, "meta.0.description", resourceName, "meta.0.description"),
-//		resource.TestCheckResourceAttrPair(dataSourceName, "meta.0.labels.key1", resourceName, "meta.0.labels.key1"),
-//		resource.TestCheckResourceAttrPair(dataSourceName, "meta.0.labels.key2", resourceName, "meta.0.labels.key2"),
-//		resource.TestCheckResourceAttrSet(dataSourceName, "meta.0.uid"),
-//	}
-//}
+func metaDataSourceAttributeCheck(dataSourceName, resourceName string) []resource.TestCheckFunc {
+	return []resource.TestCheckFunc{
+		resource.TestCheckResourceAttrPair(dataSourceName, "provisioners.0.meta.0.description", resourceName, "meta.0.description"),
```
Comment (ramya-bangera) [inline internal/resources/provisioner/provisioner_data_source_test.go:88]: By specifying  `provisioners.0.meta.0.description` does this limit the check to only one provisioners meta info even if there are multiple?

**Comment by ramya-bangera** [comment]
@shobha2626 - Does the state file dump given in description captures the state file info with LIST or GET datasource? If it's just GET can you add the LIST too? As there are changes to common meta converter code have you validated the automation against your change?

File: internal/resources/provisioner/provisioner_data_source_test.go (hunk 1 above)
Comment (shobha2626) [reply internal/resources/provisioner/provisioner_data_source_test.go:88]: In the data source config used in the test we are reading one provisioner only. Hence, we are checking for the same one here

**Comment by shobha2626** [comment]
> @shobha2626 - Does the state file dump given in description captures the state file info with LIST or GET datasource? If it's just GET can you add the LIST too? As there are changes to common meta converter code have you validated the automation against your change?

@ramya-bangera I have captured only GET because the result for LIST is huge. I have verified the LIST as well, if needed I can run again and share the same.
I have validate this change against other resource like namespace and workspace manually and there seems to be no issue. It is optional parameter and shouldn't cause any.

**Comment by ramya-bangera** [comment]
> > @shobha2626 - Does the state file dump given in description captures the state file info with LIST or GET datasource? If it's just GET can you add the LIST too? As there are changes to common meta converter code have you validated the automation against your change?
> 
> @ramya-bangera I have captured only GET because the result for LIST is huge. I have verified the LIST as well, if needed I can run again and share the same. I have validate this change against other resource like namespace and workspace manually and there seems to be no issue. It is optional parameter and shouldn't cause any.



> > @shobha2626 - Does the state file dump given in description captures the state file info with LIST or GET datasource? If it's just GET can you add the LIST too? As there are changes to common meta converter code have you validated the automation against your change?
> 
> @ramya-bangera I have captured only GET because the result for LIST is huge. I have verified the LIST as well, if needed I can run again and share the same. I have validate this change against other resource like namespace and workspace manually and there seems to be no issue. It is optional parameter and shouldn't cause any.



> > @shobha2626 - Does the state file dump given in description captures the state file info with LIST or GET datasource? If it's just GET can you add the LIST too? As there are changes to common meta converter code have you validated the automation against your change?
> 
> @ramya-bangera I have captured only GET because the result for LIST is huge. I have verified the LIST as well, if needed I can run again and share the same. I have validate this change against other resource like namespace and workspace manually and there seems to be no issue. It is optional parameter and shouldn't cause any.

Thanks for confirming , since these details was missing I just wanted to be sure that it is validated.

**Pull request merged**